frontend/src/pages/IndividualProfile.js

```diff
@@ -1,15 +1,107 @@
 import { Component } from "react";
 import { Pages } from "../globals/Enums";
+import QRCode from "react-qr-code";
+import * as OTPAuth from "otpauth";
 
 export default class IndividualProfile extends Component {
     constructor(props) {
         super(props);
-        this.state = {};
+        this.state = {
+            renderQR: false,
+            QRValue: "",
+        };
     }
+    userid = "";
+    totpSecret;
+    totp;
+    name = "";
+    phone = "";
+    email = "";
+
+    qrInterval;
+
+    componentDidMount() {
+        this.userId = localStorage.getItem("userId");
+        this.totpSecret = localStorage.getItem("totpSecret");
+
+        if (this.userId && this.totpSecret) {
+            this.name = localStorage.getItem("name");
+            this.phone = localStorage.getItem("phone");
+            this.email = localStorage.getItem("email");
+
+            this.totp = new OTPAuth.TOTP({
+                digits: 6,
+                period: 60,
+                secret: this.totpKey,
```

Comment:
Backend is expecting 8, 60 but I can change that.

Reply:
ill set frontend to 8 to match